File: Data Structures & Algorithms/is-anagram/submission-7.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False

        return collections.Counter(s) == collections.Counter(t)

        # letters_count : {str: int} = {}

        # for l in s:
        #     if l not in letters_count:
        #         letters_count[l] = 1
        #     else:
        #         letters_count[l] = letters_count[l] + 1
        
        # for l in t:
        #     if l not in letters_count:
        #         return False
        #     else:
        #         letters_count[l] = letters_count[l] - 1
        
        # for i in letters_count.values():
        #     if i != 0: return False

        # return True

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-7.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        int counter = 0;
        List<TreeNode> list = new ArrayList<>();
        TreeNode curr = root;

        while (curr != null || !list.isEmpty()) {
            while (curr != null) {
                list.add(curr);
                curr = curr.left;
            }

            curr = list.removeLast();
            counter++;

            if (counter == k) return curr.val;

            curr = curr.right;
        }

        return 0;
    }
}